src/App/App.js

```diff
@@ -2,21 +2,46 @@ import React, { Component } from 'react';
 import './App.css';
 import movieData from '../movie-data';
 import Movies from '../Movies/Movies';
+import Movie from '../Movie/Movie';
 import Header from '../Header/Header';
 
 class App extends Component {
   constructor() {
     super();
-    this.state = movieData;
+    this.state = {
+      allMovies: movieData,
+      specificMovie: null
+    };
+  }
+  displaySingleMovie = (id) => {
+    const filteredMovie = this.state.allMovies.movies.filter(movie => movie.id === id)
+    this.setState({
+      allMovies: null,
+      specificMovie: filteredMovie
```

Comment:
I see that you are updating state to toggle one of the properties to null to help you conditionally render either all movies or just the specified movie.  This is a clever approach.   As you start to implement Router, you may not have to keep resetting one of the properties in state to null because Router will let you render the appropriate things based on the URL (so you wont have to leverage that conditional logic the way you are currently).   Something to keep an eye out for as you incorporate Router